cli/src/common/pcluster_api.py

```diff
@@ -0,0 +1,165 @@
+        self.region = stack.region
+        self.version = stack.version
+        self.status = stack.status
+
+    def __repr__(self):
+        return json.dumps(self.__dict__)
+
+
+class PclusterApi:
+    """Proxy class for all Pcluster API commands used in the CLI."""
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def create_cluster(cluster_config: dict, cluster_name: str, region: str = None, disable_rollback: bool = False):
+        """
+        Load cluster model from cluster_config and create stack.
+
+        :param cluster_config: cluster configuration (yaml dict)
+        :param cluster_name: the name to assign to the cluster
+        :param region: AWS region
+        :param disable_rollback: Disable rollback in case of failures
+        """
+        try:
+            # check cluster existence
+            stack_name = get_stack_name(cluster_name)
+            if AWSApi.instance().cfn.stack_exists(stack_name):
+                return ApiFailure(f"Cluster {cluster_name} already exists.")
+
+            # Generate model from config dict and validate
+            cluster = _init_cluster_from_config(cluster_name, cluster_config)
+            validation_failures = cluster.validate()
+
+            if validation_failures:
+                # TODO skip validation errors
+                return ApiFailure(message="Configuration is invalid", validation_failures=validation_failures)
+            else:
+                # Add tags information to the cluster
+                version = get_installed_version()
+                tags = cluster.tags or []
+                tags.append(Tag(key="Version", value=version))
+                tags = [{"Key": tag.key, "Value": tag.value} for tag in tags]
+
+                # Create bucket if needed
+                cluster.setup_cluster_bucket()
+
+                creation_result = None
+                try:
+                    # Create template if not provided by the user
+                    if not (cluster.dev_settings and cluster.dev_settings.cluster_template):
+                        cluster.cluster_template_body = CDKTemplateBuilder().build_cluster_template(cluster=cluster)
+                        # print(yaml.dump(cluster.cluster_template_body))
+
+                    # upload cluster artifacts and generated template
+                    cluster.upload_artifacts()
+
+                    LOGGER.info("Creating stack named: %s", stack_name)
+                    creation_result = AWSApi.instance().cfn.create_stack(
+                        stack_name=stack_name,
+                        template_url=cluster.template_url,
+                        disable_rollback=disable_rollback,
+                        tags=tags,
+                    )
+
+                    stack = ClusterStack(cluster_name)
+                    LOGGER.debug("StackId: %s", stack.id)
+                    LOGGER.info("Status: %s", stack.status)
+                    return ClusterInfo(stack, cluster_name)
```

Comment:
Are we waiting for the stack creation complete before return the cluster info?

Reply:
Nope, the goal is having a stateless command. It means the create command starts the creation and that's all, then the user need to execute the status command and eventually create a loop to see the progress.

Reply:
I resterored the previous behaviour from the CLI, by calling multiple times the describe_cluster api call.